### DISCUSSION TITLE
How to fetch Serverside data on NextJS 12 from an external API?

### DISCUSSION THREAD

**Discussion by adrenaline681** [vercel/next.js#32691] [Help] [answered]
I want to be able to make an API call on my serverside to be able to fetch all the data required to build the page before sending it back to the client.

I used to do this using getServerSideProps inside of my page but now that getServerSideProps is not available anymore im confused on how to do this.

I have some async functions that return the fetched data from the api:

```
export async function getPrimaryData() {
    const response = await axios.get('https://example.com/primary-data/')
    return response.data
}

export async function getSecondaryData() {
    const response = await axios.get('https://example.com/secondary-data/')
    return response.data
}
```
And this is how I used to call it inside my page, for example "index.js":

```
export async function getServerSideProps(context) {
    const [primaryData, secondaryData] = await Promise.all([
        getPrimaryData(),
        getSecondaryData()
    ])

    return {props: {primaryData, secondaryData}}
}
```

Then I would simply get the data from the props of the page:

`props.primaryData`

**How can I do the same type of fetching using NextJS 12?**

I was looking through the new NextJS documentation and couldn't find anything on how to do it.

**Comment by chemicalkosek** [comment, marked as answer]
`getServerSideProps` is not deprecated. It still works in 12.
